raspi/soundserver/soundserver.py

```diff
@@ -153,6 +157,58 @@ def restore_hardvolume_route():
     vlcplayer.restore_hardvolume()
     return do_return('Ok', 200)
 
+@app.route('/reset_smart_speaker/<type>', methods=['POST','PUT'])
+def reset_smart_speaker(type):
+    current_folder = os.path.dirname(os.path.abspath(__file__))
+    wap_script = os.path.abspath(current_folder + '/../access_point/wap.sh')
+    factory_reset = os.path.abspath(current_folder + '/../factory_reset/factory_reset.sh')
+    if type == 'hard' :
+        logger.info("hard FACTORY RESET")
+        logger.info("hard factory reset initiated")
+        subprocess.Popen(['sudo','bash', factory_reset, 'hard'])
+    elif type == 'soft' :
+        logger.info("soft FACTORY RESET")
+        logger.info("soft factory reset initiated")
+        subprocess.Popen(['sudo','bash', factory_reset, 'soft'])
+    elif type == 'AP' :
+        logger.info("switch to access point mode")
+        logger.info("switch to access mode initiated")
+        subprocess.Popen(['sudo','bash', wap_script])  
+    return do_return('Ok', 200)      
+
+@app.route('/getdevice', methods=['GET'])
+def get_mounted_device():
+    folders = os.listdir(mountPath)
+    devices = []
+    for d in folders:
+            device = {}
+            device['name'] = d
+            devices.append(device)
+                
+    return do_return(devices, 200)
+
+@app.route('/getOfflineSong/<folder>', methods=['GET'])
+def get_offline_song(folder):
+    files = os.listdir(mountPath+folder)
+    songs = []
+    for i in files:
+	    if i.endswith('.mp3') or i.endswith('.m4a') or i.endswith('.ogg') or i.endswith('.deb'):
+                file = {}
+                file['name'] = i
+                songs.append(file)
+    return do_return(songs, 200)
+
+@app.route('/playOfflineSong/<folder>/<file>', methods=['PUT'])
+def play_offine_song(folder,file):
+    vlcplayer.play(mountPath+'/'+folder+'/'+file)
```

Comment:
There must be whitespaces, following PEP8.

To build the path, should use `os.path` or `pathlib`, not concatenate strings like this.